src/adapters/claude/helpers/completions.ts

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable sonarjs/no-duplicate-string */
+
+import Anthropic from "@anthropic-ai/sdk";
+import { Context } from "../../../types";
+import { ContentBlock } from "@anthropic-ai/sdk/resources";
+import { SuperAnthropic } from "./claude";
+
+export interface CompletionsType {
+  answer: string;
+  groundTruths: string[];
+  tokenUsage: {
+    input: number;
+    output: number;
+    total: number;
+  };
+}
+
+// Type guard for content block
+interface TextBlock {
+  type: "text";
+  text: string;
+}
+
+function isTextBlock(content: ContentBlock): content is TextBlock {
+  return content?.type === "text" && typeof content?.text === "string";
+}
+
+export class AnthropicCompletion extends SuperAnthropic {
+  constructor(client: Anthropic, context: Context) {
+    super(client, context);
+  }
+
+  getModelMaxTokenLimit(model: string): number {
+    const tokenLimits = new Map<string, number>([["claude-3.5-sonnet", 200000]]);
+
+    return tokenLimits.get(model) || 200000;
```

Comment:
I think having a default of 200k is not a good idea. It should throw an error if the number isn't found in our lookup. 

200k only corresponds to a specific model that we are using now.